File: src/priv/bajdcc/util/lexer/regex/Constructure.java
package priv.bajdcc.util.lexer.regex;

import java.util.ArrayList;

/**
 * 顺序结构或分支结构
 * 
 * @author bajdcc
 */
public class Constructure implements IRegexComponent {
	/**
	 * 若是则为分支，否则为顺序
	 */
	public boolean bBranch = false;

	/**
	 * 孩子结点
	 */
	public ArrayList<IRegexComponent> arrComponents = new ArrayList<IRegexComponent>();

	public Constructure(boolean branch) {
		bBranch = branch;
	}

	@Override
	public void visit(IRegexComponentVisitor visitor) {
		visitor.visitBegin(this);
		for (IRegexComponent component : arrComponents) {
			component.visit(visitor);
		}
		visitor.visitEnd(this);
	}
}

File: src/main/java/com/bajdcc/util/lexer/regex/Constructure.kt
package com.bajdcc.util.lexer.regex

/**
 * 顺序结构或分支结构
 * [branch] 若是则为分支，否则为顺序
 * @author bajdcc
 */
class Constructure(var branch: Boolean) : IRegexComponent {

    /**
     * 孩子结点
     */
    var arrComponents = mutableListOf<IRegexComponent>()

    override fun visit(visitor: IRegexComponentVisitor) {
        visitor.visitBegin(this)
        arrComponents.forEach { it.visit(visitor) }
        visitor.visitEnd(this)
    }
}
